src/backend/distributed/connection/locally_reserved_shared_connections.c

```diff
@@ -377,7 +377,7 @@ EnsureConnectionPossibilityForNodeList(List *nodeList)
 /*
  * EnsureConnectionPossibilityForNode reserves a shared connection
  * counter per node in the nodeList unless:
- *  - Reservation is possible/allowed (see IsReservationPossible())
+ *  - Reservation is not possible/allowed (see IsReservationPossible())
```

Comment:
why `not`? This function does nothing when reservation is not possible.

Reply:
The `not` cancels the `unless` in the sentence before. Without it it reads like:
> ... reserves a shared connection unless reserving is possible/allowed